### DISCUSSION TITLE
A touch of help for a newbie if possible

### DISCUSSION THREAD

**Discussion by OriginBeta** [community/community#124424] [Programming Help]
### Select Topic Area

Bug

### Body

Hello all, 

total newbie but working as hard as I can to learn. seem to have made the mistake of hoping chat gpt could help line me up and instead it appears I am now worse off than before. I could really use some help getting the base functionality to work in this. Its a super simple app and wont ever earn anything lol

https://github.com/OriginBeta/Vulcan-Husbandry/blob/fac77c855e52f16708592478c790e7b52ba5008c/.github/workflows/objective-c-xcode.yml

**Comment by ebndev** [comment]
Hey @OriginBeta - can you provide more context on what you're trying to accomplish and what errors you're running into?

Giving a few more details might help someone give you a nudge in the right direction!

**Comment by OriginBeta** [comment]
So the basics are simple the app is used to track animal husbandry data on one tab, and contains the basics of care for numerous animals on another. The idea is that the user can drill down on a data tab to locate said data, additionally when the user is tracking, I am trying to get this force the user to select the animal by unique name and then track data within that. Ideally it will pull from the animal data to include this care data on the animal tracking page

**Comment by RafaelJohn9** [comment]
@OriginBeta I'll make a pull request
you can check the code if it matches the requirements if done to expectation, else you can just communicate back

**Comment by RafaelJohn9** [comment]
@OriginBeta  I'll send the pull request when am done

**Comment by dcaayushd** [comment]
Based on the provided code, there are a few errors that need to be addressed:

1. **Missing initializer for 'Animal' to satisfy protocol 'Hashable'**

To fix this error, you need to implement the `Hashable` protocol's requirements for the `Animal` struct. The `Hashable` protocol requires you to implement the `hash(into:)` method and the `==` static function. Here's how you can do it:

```swift
struct Animal: Hashable {
    // ...

    func hash(into hasher: inout Hasher) {
        hasher.combine(commonName)
    }

    static func ==(lhs: Animal, rhs: Animal) -> Bool {
        return lhs.commonName == rhs.commonName
    }
}
```

2. **Type '([String : [Animal]])' cannot conform to 'View'**

The error occurs because you're trying to pass an instance of `[String: [Animal]]` (a dictionary) directly to the `MainView` struct, which expects a `View` instance. To fix this, you need to wrap the dictionary in a `@State` property or pass it as a parameter to the `MainView` struct.

Here's how you can pass it as a parameter:

```swift
@main
struct ReptileTrackingApp: App {
    var body: some Scene {
        WindowGroup {
            MainView(categories: ["Lizards", "Amphibians", "Snakes"], reptilesAndAmphibians: reptilesAndAmphibians)
        }
    }
}
```

3. **Type '(Animal) -> ()' cannot conform to 'View'**

This error occurs because you're trying to pass an instance of `Animal` directly to the `CareDetailView` struct, which expects a `View` instance. To fix this, you need to pass the `Animal` instance as a parameter to the `CareDetailView` struct.

Here's how you can do it:

```swift
CareDetailView(animal: selectedAnimal)
    .padding()
```

4. **Cannot convert value of type 'String' to expected argument type 'Animal?'**

This error occurs because you're trying to assign a `String` value to the `$selectedAnimal` binding, which expects an `Animal?` value. To fix this, you need to find the corresponding `Animal` instance from the `reptilesAndAmphibians` dictionary based on the selected category and common name.

Here's an example of how you can do it:

```swift
Picker("Select Animal", selection: $selectedAnimal) {
    ForEach(animalsInCategory, id: \.self) { animal in
        Text(animal.commonName)
            .tag(Optional(animal))
    }
}
```

After making these changes, your code should compile without any errors.

**Reply by OriginBeta** [reply]
Thank you, I appreciate the assistance. I will get to work on these changes and let you know if anything manages to go awry.
